core/src/main/java/org/elasticsearch/index/query/WrapperQueryBuilder.java

```diff
@@ -86,4 +92,44 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        try (XContentParser qSourceParser = XContentFactory.xContent(source).createParser(source)) {
+            final QueryParseContext parseContext = new QueryParseContext(context);
+            parseContext.reset(qSourceParser);
+            Query query = parseContext.parseInnerQuery();
+            context.combineNamedQueries(context);  // NO COMMIT: not really sure if this is the right way
```

Comment:
Yes, I think you are right. Previously in the parser, the context to parse the wrapped query was a copy with empty namedQueries, to whatever got added there during the inner parsing needed to be re-added to the actual context. We now add all the named queries in AbstractQueryBuilder#toQuery(), so I think when we use the original shard context here, all named queries generated by the inner query builder are added there too.